src/library_manager/lib_manager.c

```diff
@@ -529,8 +530,8 @@ static void __sparse_cache *lib_manager_allocate_store_mem(uint32_t size,
 		return NULL;
 	}
 
-	sys_cache_data_invd_range(local_add, size);
-	sys_cache_instr_invd_range(local_add, size);
+	dcache_invalidate_region(local_add, size);
+	icache_invalidate_region(local_add, size);
```

Comment:
Why we'd use XTOS wrappers in code that is already Zephyr native (this is including <zephyr/cache.h>)...?

Reply:
@kv2019i because our wrappers take correct sparse address space as an argument. @ujfalusi initially proposed adding casts there too, but I think it's better to use our wrappers (we also use them a lot everywhere) without type-casting - where possible

Reply:
right, that's murky but I'll buy the argument. We shouldn't then be including izephyr/cache.h anymore right?

Reply:
@kv2019i yeah, I think we can remove that header now. And BTW there are already multiple calls to `dcache_*_region()` in this file, so, yes, I should've stuck with them, sorry.